### PR TITLE
v1.4: Bug-fix/Update ingredient name to be case-insensitive

### LINKED ISSUE #224: [PE Dry Run]Delete command does not seems to work properly
These are the two items that are the same name but different case lettering.

image

When executed this command by this specific name, it deleted the first entry. 

image

Bug: If the delete of name is case-sensitive, neither item should be deleted
if the delete of name is case-insensitive, both item should be deleted.

Also, the deleted item name does not match the exact name of the deleted item

<hr>

**Reported by:** @crimsonang
**Severity:** `Medium`

<sub>[original: nus-cs2103-AY1819S1/pe-dry-run#319]</sub>

### PR DESCRIPTION
**Pull request by rebstan97**
Closes #224, #232, #235, #242
"Tomato" is now the same as ingredient as "TOMATO" or "toMAto" etc and cannot be concurrently contained in the ingredient list.

### CODE REVIEW

**Comment by AZhiKai** [comment]
Do add more tests to test those case sensitivity issue.

**Comment by rebstan97** [comment]
> Do add more tests to test those case sensitivity issue.

I have added one case for lower case only, one for upper case only, and one for a mix of upper and lower cases.

**Comment by AZhiKai** [comment]
Thats sounds like it covers it all. :+1:

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -385,15 +385,16 @@ Examples:
 * `add-item n/Burger p/2`
 * `ai n/Burger Set p/4.5 t/Set`
 
-==== Listing all items in the menu : `list-items` or `li`
+==== Listing all items in the menu : `list-items` or 'li'
```
Comment (yican95) [inline docs/UserGuide.adoc]: Not sure why but the UserGuide is reverting all my changes for menu management

Comment (AZhiKai) [reply docs/UserGuide.adoc]: I think rebecca hasn't updated her branch so her UG version is outdated.

Comment (rebstan97) [reply docs/UserGuide.adoc]: Not sure what happened but I have reverted the commit and added my own changes. @yican95 please check if it is okay now. Thanks for pointing out!

**Pull request merged**